### PR TITLE
Add numa variant in ucx package

### PR DESCRIPTION
**Pull request by tbhaxor**
- enable / disable numa based on variant

### CODE REVIEW

**Comment by haampie** [comment]
1. Why 8cf032d9f6df35d440c43a04da422604c1c22bad? (the commit hash ends with `bad` :laughing: )

2. numa support is removed in https://github.com/openucx/ucx/pull/8853 (UCX 1.15.0)

File: var/spack/repos/builtin/packages/ucx/package.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ class Ucx(AutotoolsPackage, CudaPackage):
         description="Build shared libs, static libs or both",
     )
     variant("logging", default=False, description="Enable logging")
+    variant("numa", default=False, description="Enable NUMA support")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/ucx/package.py]:
```suggestion
    variant("numa", default=False, when="@:1.14", description="Enable NUMA support")
```

File: var/spack/repos/builtin/packages/ucx/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/ucx/package.py:1]: Please add back `depends_on("numactl", when="+numa")`

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/ucx/package.py (hunk 1 above)
Comment (tbhaxor) [reply var/spack/repos/builtin/packages/ucx/package.py]: @haampie Would you help explain version ranges like this? I see it everywhere in packages but hard to understand what does it mean actually.

Comment (tbhaxor) [reply var/spack/repos/builtin/packages/ucx/package.py]: So what I understood it will apply for all the versions `<= 1.14.0`

Comment (tbhaxor) [reply var/spack/repos/builtin/packages/ucx/package.py]: Also let's not add `when` condition for now, as its unclear to me why you missed `v1.14.1`

File: var/spack/repos/builtin/packages/ucx/package.py
Comment (tbhaxor) [reply var/spack/repos/builtin/packages/ucx/package.py:1]: Done

**Comment by zzzoom** [comment]
Shouldn't `+numa` be the default to maintain the former behavior?

File: var/spack/repos/builtin/packages/ucx/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/ucx/package.py]: See https://spack.readthedocs.io/en/latest/basic_usage.html#version-specifier

`@:1.14` includes 1.14.x but excludes 1.15 and above, please add it so we don't forget it on the 1.15 release.

**Comment by haampie** [comment]
Makes sense to go for the least breaking change of  `default=True` for the `numa` variant

**Comment by tbhaxor** [comment]
> 

For versions less than 1.15?

keeping "numa support is removed in https://github.com/openucx/ucx/pull/8853 (UCX 1.15.0)" in mind

**Review by haampie — DISMISSED** [review]

File: var/spack/repos/builtin/packages/ucx/package.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ class Ucx(AutotoolsPackage, CudaPackage):
         description="Build shared libs, static libs or both",
     )
     variant("logging", default=False, description="Enable logging")
+    variant("numa", default=True, description="Enable NUMA support")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/ucx/package.py]:
```suggestion
    variant("numa", default=True, when="@:1.14", description="Enable NUMA support")
```

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/ucx/package.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ class Ucx(AutotoolsPackage, CudaPackage):
         description="Build shared libs, static libs or both",
     )
     variant("logging", default=False, description="Enable logging")
-    variant("numa", default=True, description="Enable NUMA support")
+    variant("numa", default=True, when="@:1.14", description="Enable NUMA support")
```
Comment (tbhaxor) [inline var/spack/repos/builtin/packages/ucx/package.py:75]: This change is not clear to me @haampie

**Pull request merged**